src/Parser.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Worksome\Number;
+
+use BCMath\Number as BCNumber;
+use InvalidArgumentException;
+
+class Parser
+{
+    /**
+     * Parse the given number and extract the whole number and decimal number
+     *
+     * @return array<int, string>
+     */
+    public static function parseFragments(Number|BCNumber|string|int|float $num): array
+    {
+        $num = (string) $num;
+        $pos = strpos($num, Number::DECIMAL_SYMBOL);
+        $wholeNumber = $num;
+        $decimalNumber = '';
+
+        if ($pos !== false) {
+            $wholeNumber = substr($num, 0, $pos);
+            $decimalNumber = substr($num, $pos + 1);
+        }
+
+        return [
+            $wholeNumber,
+            $decimalNumber,
+        ];
+    }
+
+    /**
+     * Convert the given float to its string form
+     */
+    public static function floatToString(float $number, null|int $precision = null): string
```

Comment:
Can you not just use `number_format()`?

Reply:
Yeah I don't know. I did a bit of research and wasn't able to find a reliable way to convert a float to string. 

The `number_format` function requires a `decimals` argument which, when too precise (e.g. input is 1.2 and you use a precision of 320), produces an odd output (e.g. 1.199...)

Feel free to play around - I added `FloatToStringTest` to pick up when things break. Not sure how perfect the datasets are (specifically the PHP_FLOAT_MIN/PHP_FLOAT_MAX ones) 🤷

Reply:
Temporarily reverted it back to `(string) $float`. Unsure how suitable this is though 🤷